src/book/02-system/01-nubots/03-continuous-integration.mdx

```diff
@@ -0,0 +1,134 @@
+
+</Alert>
+
+### Install Docker
+
+Steps from <https://docs.docker.com/install/linux/docker-ce/ubuntu/>.
+
+```bash
+# Remove old Docker
+sudo apt-get remove docker docker-engine docker.io containerd runc
+
+# Install packages to allow apt to use a repository over HTTPS
+sudo apt-get update
+sudo apt-get install apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+
+# Add Docker’s official GPG key
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+
+# Add Docker's stable repository
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+​
+# Install latest Docker CE
+sudo apt-get update
+sudo apt-get install docker-ce docker-ce-cli containerd.io
+```
+
+### Install and Configure Buildkite
+
+Steps from <https://buildkite.com/docs/agent/v3/ubuntu>.
+
+```bash
+# Add the signed Buildkite apt repository
+echo "deb https://apt.buildkite.com/buildkite-agent stable main" | sudo tee /etc/apt/sources.list.d/buildkite-agent.list
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 32A37959C2FA5C3C99EFBC32A79206696452D198
+
+# Install Buildkite the agent
+sudo apt-get update
+sudo apt-get install -y buildkite-agent
+
+# Add the agent token to the configuration file.
+# Replace INSERT-YOUR-AGENT-TOKEN-HERE with the token from Buildkite.
+sudo sed -i "s/xxx/INSERT-YOUR-AGENT-TOKEN-HERE/g" /etc/buildkite-agent/buildkite-agent.cfg
+
+# Start the agent and schedule it to run on system startup
+sudo systemctl enable buildkite-agent && sudo systemctl start buildkite-agent
+```
+
+### Configure Docker
+
+Steps from <https://docs.docker.com/install/linux/linux-postinstall/>.
+
+```bash
+# Create a 'docker' user group and add local users to allow for use without sudo
+sudo groupadd docker                     # Create user group
+sudo usermod -aG docker nubots           # Add nubots to the group
+sudo usermod -aG docker buildkite-agent  # Add buildkite-agent to the group
+newgrp docker                            # Activate group changes
+
+# Configure Docker to run on boot
+sudo systemctl enable docker
+```
+
+### Add DockerHub Credentials to Buildkite
+
+For agents to push built `master` images to DockerHub, they need the service account login credentials. The username of this account is `nubotsdocker`, and is specified in the Buildkite [pipeline file](https://github.com/NUbots/NUbots/blob/e7db7154a689ab076041375894b3baf1d53a2587/.buildkite/pipeline.yml#L11). The password needs to be made available to the agent when it runs, via the `DOCKER_LOGIN_PASSWORD` env variable.
+
+One way to provide credentials to the agent via env variables is to use the agent hooks system. The default directory for agent hooks is `/etc/buildkite-agent/hooks`. The default path for the agent's configuration file is `/etc/buildkite-agent/buildkite-agent.cfg`.
```

Comment:
Rather than `one way` I think it should just be
```suggestion
Credentials are provided to the agent via env variables from agent hooks system. Specifically via the environment hook. The default location for this hook is `/etc/buildkite-agent/hooks/environment`. The default path for the agent's configuration file is `/etc/buildkite-agent/buildkite-agent.cfg`.
```